Bottom: a622ecbfd685a6243706e776325d28ca4e1b9cea
Top:    65711857d77476acdd3ace39d826e8e2af9137b2
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-30 05:40:01 +0100

Refresh of convert-object-memory-allocati

---

diff --git a/src/misc.c b/src/misc.c
index efca2e1..cbbb172 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -1703,8 +1703,11 @@ GetGridLockCoordinates (int type, void *ptr1,
       *y = ((ElementTypePtr) ptr2)->MarkY;
       break;
     case POLYGON_TYPE:
-      *x = ((PolygonTypePtr) ptr2)->Points[0].X;
-      *y = ((PolygonTypePtr) ptr2)->Points[0].Y;
+      {
+        PointType *point0 = ((PolygonTypePtr) ptr2)->Points->data;
+        *x = point0->X;
+        *y = point0->Y;
+      }
       break;
 
     case LINEPOINT_TYPE:
